srcCpp/Infrastructure_pro.cxx

```diff
@@ -646,6 +741,83 @@ bool configureShmemTransport(
+            + ".parent.message_size_max";
+
+    DDS_Property_t *parentProp = DDSPropertyQosPolicyHelper::lookup_property(
+            qos.property,
+            propertyName.c_str());
+    if (parentProp != NULL && parentProp->value != NULL) {
+        // printf("Value for %s is %s\n", propertyName.c_str(), parentProp->value);
+        return atoi(parentProp->value);
+    } else {
+        // printf("Value for %s not found, returning default\n",propertyName.c_str());
+        return DEFAULT_MESSAGE_SIZE_MAX;
+    }
+}
+
+/*
+ * Configures the minimumMessageSizeMax value in the PerftestTransport object with
+ * the minimum value for all the enabled transports in the XML configuration.
+ */
+void getTransportMinimumMessageSizeMax(
+        PerftestTransport &transport,
+        DDS_DomainParticipantQos& qos)
+{
+    long qosConfigurationMessageSizeMax = MESSAGE_SIZE_MAX_NOT_SET;
+    long transportMessageSizeMax = MESSAGE_SIZE_MAX_NOT_SET;
+    if ((qos.transport_builtin.mask & DDS_TRANSPORTBUILTIN_SHMEM) != 0) {
+        transportMessageSizeMax = getTransportMessageSizeMax("SHMEM", transport, qos);
+        if (transportMessageSizeMax < qosConfigurationMessageSizeMax) {
+            qosConfigurationMessageSizeMax = transportMessageSizeMax;
+        }
+    }
+    if ((qos.transport_builtin.mask & DDS_TRANSPORTBUILTIN_UDPv4) != 0) {
+        transportMessageSizeMax = getTransportMessageSizeMax("UDPv4", transport, qos);
+        if (transportMessageSizeMax < qosConfigurationMessageSizeMax) {
+            qosConfigurationMessageSizeMax = transportMessageSizeMax;
+        }
+    }
+    if ((qos.transport_builtin.mask & DDS_TRANSPORTBUILTIN_UDPv6) != 0) {
+        transportMessageSizeMax = getTransportMessageSizeMax("UDPv6", transport, qos);
+        if (transportMessageSizeMax < qosConfigurationMessageSizeMax) {
+            qosConfigurationMessageSizeMax = transportMessageSizeMax;
+        }
+    }
+    if (transport.transportConfig.kind == TRANSPORT_TCPv4
+            || transport.transportConfig.kind == TRANSPORT_TLSv4) {
+        transportMessageSizeMax = getTransportMessageSizeMax("TCP", transport, qos);
+        if (transportMessageSizeMax < qosConfigurationMessageSizeMax) {
+            qosConfigurationMessageSizeMax = transportMessageSizeMax;
+        }
+    }
+    if (transport.transportConfig.kind == TRANSPORT_DTLSv4) {
+        transportMessageSizeMax = getTransportMessageSizeMax("DTLS", transport, qos);
+        if (transportMessageSizeMax < qosConfigurationMessageSizeMax) {
+            qosConfigurationMessageSizeMax = transportMessageSizeMax;
+        }
+    }
+    if (transport.transportConfig.kind == TRANSPORT_WANv4) {
+        transportMessageSizeMax = getTransportMessageSizeMax("WAN", transport, qos);
+        if (transportMessageSizeMax < qosConfigurationMessageSizeMax) {
+            qosConfigurationMessageSizeMax = transportMessageSizeMax;
+        }
+    }
+
+    transport.minimumMessageSizeMax = transportMessageSizeMax;
+}
```

Comment:
Because UDPv4, UDPv6 and SHMEM are the builtin masks that can be alone or combined between them, I mean like:

UDPv4||SHMEM or UDPv4||UDPv6 or UDPv4||UDPv6||SHMEM